### PR TITLE
Text Input

### PR DESCRIPTION
**Pull request by rfisher58**
Text input has the functionality per figma and jira, could use polishing. Current issues:

1. When multiline is triggered you are taken out of the input and have to re-click into it to continue typing. 
2. When multiline is triggered the clearable X remains in the middle of the input as does the left side icon.

### CODE REVIEW

File: packages/hs-react-ui/src/components/Text/Text.stories.tsx [hunk 1]
```diff
@@ -26,6 +26,9 @@ const themeTimings = {
 
 const icon = <Icon path={Icons.mdiComment} size='1rem' />;
 
+const getIconPrefixPath = (path: string) => path ? <Icon size={'16px'} path={path} /> : null
+const getIconSuffixPath = (path: string) => path ? <Icon size={'16px'} path={path} /> : null
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Text/Text.stories.tsx:30]: could just be one function called twice

File: packages/hs-react-ui/src/components/Text/Text.stories.tsx [hunk 2]
```diff
@@ -44,8 +47,8 @@ export const Default = () => (
   <Text
     size={text('size', '1rem')}
     color={color('color', colors.grayXlight)}
-    iconPrefix={select('iconPrefix', options, options.mdiComment)}
-    iconSuffix={select('iconSuffix', options, options.mdiComment)}
+    iconPrefix={getIconPrefixPath(select('iconPrefix', options, options.mdiComment))}
+    iconSuffix={getIconSuffixPath(select('iconPrefix', options, options.mdiComment))}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Text/Text.stories.tsx:51]: looks like this change breaks the iconSuffix knob
```suggestion
    iconSuffix={getIconSuffixPath(select('iconSuffix', options, options.mdiComment))}
```

File: packages/hs-react-ui/src/components/Text/Text.stories.tsx [hunk 3]
```diff
@@ -81,8 +84,8 @@ export const ThemedText = () => (
     StyledContainer={ThemedContainer}
     size={text('size', '1rem')}
     color={color('color', colors.grayDark)}
-    iconPrefix={select('iconPrefix', options, options.none)}
-    iconSuffix={select('iconSuffix', options, options.none)}
+    iconPrefix={getIconPrefixPath(select('iconPrefix', options, options.mdiComment))}
+    iconSuffix={getIconSuffixPath(select('iconPrefix', options, options.mdiComment))}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/Text/Text.stories.tsx:88]: again this knob should be iconSuffix
```suggestion
    iconSuffix={getIconSuffixPath(select('iconSuffix', options, options.mdiComment))}
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  rows?: any,
+  errorMessage?: string,
+  style?: any;
+}
+
+// const TextInputContainers = {
+//   [TextInputTypes.icon]: IconInputContainer
+// };
+
+
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:71]:
```suggestion
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  const getIconPath = (path: string) => path ? <Icon size={'16px'} path={path} /> : null
+  
+  const isMultiLine = (inputValue: string) => (inputValue.length > 15) ? true : false;
+
+  const isError = (inputValue: string) => (inputValue.length < 5 && inputValue.length > 0) ? 'Short Message Error' : null;
+
+  return (
+
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx:35]: Could condense these lines a bit
```suggestion
  const [inputValue, setInputValue] = useState('');
  const options = {
    none: '',
    ...IconPaths
  };
  const getIconPath = (path: string) => path ? <Icon size={'16px'} path={path} /> : null
  const isMultiLine = (inputValue: string) => (inputValue.length > 15) ? true : false;
  const isError = (inputValue: string) => (inputValue.length < 5 && inputValue.length > 0) ? 'Short Message Error' : null;
  
  return (
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  const options = {
+    none: '',
+    ...IconPaths
+  };
+
+  const getIconPath = (path: string) => path ? <Icon size={'16px'} path={path} /> : null
+  
+  const isMultiLine = (inputValue: string) => (inputValue.length > 15) ? true : false;
+
+  const isError = (inputValue: string) => (inputValue.length < 5 && inputValue.length > 0) ? 'Short Message Error' : null;
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx:32]: I love these as examples of the power of the component - but it should probably be put in a separate story, so people don't think that this behavior comes with the TextInput out of the box.

Also the error message here could be "Text must be longer than 4 characters"

File: packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      setInputValue(event.target.value)
+      action('onChange')(event.target.value)}}
+    value={inputValue}
+    placeholder={text('placeholder', 'Place Holder')} 
+    onClear={(event: SyntheticEvent) => {
+      setInputValue('')
+      action('onClear')()}}
+    iconPrefix={getIconPath(select('iconPrefix', options, options.mdiComment))}
+    multiline={isMultiLine(inputValue)}
+    errorMessage={isError(inputValue)}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx:47]: this just needs one more level of indention

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+const TextInput = ({
+  placeholder = 'test props',
+  iconPrefix,
+  onClear,
+  onChange,
+  value,
+  multiline,
+  errorMessage,
+  style, 
+
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:81]:
```suggestion
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+}: TextInputProps) => {
+  const TextInput = TextInputContainer
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:83]: This should be moved into the props! Check out the other components (like `Card`) to see how the sub-component props allow the end-user/developer to overwrite our styled-components with their own.

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+  return (
+    <DivContainer>
+      {<IconContainer>{iconPrefix}</IconContainer>}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:86]: It looks like you were going to add a conditional at the start of this
```suggestion
      {iconPrefix && <IconContainer>{iconPrefix}</IconContainer>}
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+      {onClear && value && <ClearIconContainer onClick={onClear}><Icon path={mdiClose} size='16px' /></ClearIconContainer>}
+      <TextInput as={multiline ? 'textarea' : 'input'} hasMultiline={!!multiline} hasIconPrefix={!!iconPrefix} placeholder={placeholder} onChange={onChange} value={value} />
+      {<ErrorContainer>{errorMessage}</ErrorContainer>}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:89]: same here
```suggestion
      {errorMessage && <ErrorContainer>{errorMessage}</ErrorContainer>}
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
+import React, { SyntheticEvent } from 'react';
+import styled from 'styled-components';
+import Icon from '@mdi/react';
+import { mdiClose } from '@mdi/js';
+
+import { TextInputTypes } from '../../enums/TextInputTypes';
+
+
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:8]: It doesn't look like this is being used
```suggestion
```

File: packages/hs-react-ui/src/enums/TextInputTypes.ts [hunk 13]
```diff
@@ -0,0 +1,3 @@
+export enum TextInputTypes {
+    icon = 'icon'
+}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/enums/TextInputTypes.ts:3]: looks like this isn't being used - let's delete the file

File: packages/hs-react-ui/src/components/Text/Text.stories.tsx (hunk 1 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/components/Text/Text.stories.tsx]: Resolved

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx (hunk 4 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx]: Resolved

File: packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx (hunk 5 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/components/TextInput/TextInput.stories.tsx]: Resolved

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx (hunk 12 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx]: resolved

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx (hunk 9 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx]: resolved

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx (hunk 10 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx]: resolved

File: packages/hs-react-ui/src/enums/TextInputTypes.ts (hunk 13 above)
Comment (rfisher58) [reply packages/hs-react-ui/src/enums/TextInputTypes.ts]: resolved

**Pull request merged**